cases/security_test.py

```diff
@@ -154,48 +206,88 @@ async def test_ssp(
-                assert_equal(wait_res.result_variant(), 'connection')
-                assert init_res.connection is not None and wait_res.connection is not None
-                return init_res.connection, wait_res.connection
+            nonlocal ref_dut_bumble
+            nonlocal dut_ref_bumble
+            nonlocal ctkd_task
 
             # Make classic connection.
             if connect == 'incoming_connection':
-                ref_dut, dut_ref = await bredr_connect(self.ref, self.dut)
+                ref_dut, dut_ref = await make_bredr_connection(initiator=self.ref, acceptor=self.dut)
             else:
-                dut_ref, ref_dut = await bredr_connect(self.dut, self.ref)
+                dut_ref, ref_dut = await make_bredr_connection(initiator=self.dut, acceptor=self.ref)
 
+            # Retrieve Bumble connection
+            if isinstance(self.dut, BumblePandoraDevice):
+                dut_ref_bumble = self.dut.device.lookup_connection(int.from_bytes(dut_ref.cookie.value, 'big'))  # type: ignore
             # Role switch.
             if isinstance(self.ref, BumblePandoraDevice):
-                ref_dut_raw = self.ref.device.lookup_connection(int.from_bytes(ref_dut.cookie.value, 'big'))
-                if ref_dut_raw is not None:
+                ref_dut_bumble = self.ref.device.lookup_connection(int.from_bytes(ref_dut.cookie.value, 'big'))  # type: ignore
+                if ref_dut_bumble is not None:
                     role = {
                         'against_central': HCI_CENTRAL_ROLE,
                         'against_peripheral': HCI_PERIPHERAL_ROLE,
                     }[ref_role]
 
-                    if ref_dut_raw.role != role:
+                    if ref_dut_bumble.role != role:
                         self.ref.log.info(
                             f"Role switch to: {'`CENTRAL`' if role == HCI_CENTRAL_ROLE else '`PERIPHERAL`'}"
                         )
-                        await ref_dut_raw.switch_role(role)
+                        await ref_dut_bumble.switch_role(role)
+
+            async def wait_ctkd_keys() -> List[PairingKeys]:
+                futures: List[asyncio.Future[PairingKeys]] = []
+                if ref_dut_bumble is not None:
+                    ref_dut_fut = asyncio.get_event_loop().create_future()
+                    futures.append(ref_dut_fut)
+
+                    def on_pairing(keys: PairingKeys) -> None:
+                        ref_dut_fut.set_result(keys)
+
+                    ref_dut_bumble.on('pairing', on_pairing)
+                if dut_ref_bumble is not None:
+                    dut_ref_fut = asyncio.get_event_loop().create_future()
+                    futures.append(dut_ref_fut)
+
+                    def on_pairing(keys: PairingKeys) -> None:
+                        dut_ref_fut.set_result(keys)
+
+                    dut_ref_bumble.on('pairing', on_pairing)
+
+                return await asyncio.gather(*futures)
```

Comment:
@uael @DeltaEvo 
Since our target is to avoid direct usage of Bumble, do you think it's possible to change the behavior of `pandora.host.Secure(BR_EDR_CONNECTION, LE_LEVEL)` to CTKD(and so does `WaitSecurity`)?

Reply:
That's a good question 🤔 
Actually, using bumble internals for this kind of tests is intended since it's a specific behavior BUT in the meantime it also makes sense to ctkd if possible when `Secure`/`WaitSecurity`.
I would say that it's fine for now as it's easier and clearer to develop the test without hidden flow control, what do you think @DeltaEvo ?

Reply:
Keep this opened, but it's not urgent.